storagemarket/impl/provider.go

```diff
@@ -342,6 +347,10 @@ func (p *Provider) ImportDataForDeal(ctx context.Context, propCid cid.Cid, data
 		return xerrors.Errorf("given data does not match expected commP (got: %s, expected %s)", pieceCid, d.Proposal.PieceCID)
 	}
 
+	log.Debugw("will fire ProviderEventVerifiedData for imported file", "propCid", propCid)
+	if err := tempfi.Close(); err != nil {
+		log.Errorw("failed to close temp imported local file", "propCid", propCid, "err", err)
+	}
```

Comment:
Maybe move this in `cleanup` ?

Reply:
It looks redundant to both defer a close and close explicitly here. The deferred close should be enough.

Reply:
Removed redundant close.